### DISCUSSION TITLE
How to display Hyperlinks in Search Results

### DISCUSSION THREAD

**Discussion by szyvett** [microsoft-search/pnp-modern-search#2062] [Q&A] [answered]
Hi all, 

I am still just getting to learn about PnP Search and anything that is a little advanced goes above my head. I want to display items from a SharePoint List Hyperlink column in the PnP Search Results web part that displays as the display name and if I click on it, it would send me to the link contained in the hyperlink item. 
![chrome_kfJ5COO1dx](https://user-images.githubusercontent.com/32570227/166666137-ebf76df9-13c4-4fe6-8129-e826fd8f6f1f.png)
![chrome_zr7wcliDv1](https://user-images.githubusercontent.com/32570227/166666711-cb205524-9f81-4344-9813-44ec11c78ff9.png)

I am having struggles finding the right managed properties for the display name and the link. Could someone help me out please? 

Many thanks in advance, 
Yvett

**Comment by FranckyC** [comment, marked as answer]
Hi @szyvett,

Hyperlink columns corresponds to `<column_name>OWSURLH` search managed properties so you should add those properties to the 'Selected properties' list in the SharePoint Search source configuration:

![image](https://user-images.githubusercontent.com/5659122/166679893-6e657586-8003-466c-a737-68e5d0f97e0a.png)

Then, in the layout you want to use that property, you must use the Handlebars helper function `{{getUrlField MyLinkOWSURLH "Title"}}` to get the title of the the link or `{{getUrlField MyLinkOWSURLH "URL"}}` for the url.

![image](https://user-images.githubusercontent.com/5659122/166680327-85da9cca-67af-4dff-a392-e524acb832f6.png)

Hope it helps.

**Reply by szyvett** [reply]
Hi @FranckyC , 

Thank you for your quick reply! I have done everything as you have instructed me, and it looks exactly as I wanted it, however, when I clicked on the link for Google for example, instead of going to https://www.google.com, the browser was redirected to `https://www.[tenant].sharepoint.com/sites/[currentSite]/Google` even though I have specified the href to be `{{getUrlField MyLinkOWSURLH "URL"}}` as below. 
![chrome_xrDxPfzshw](https://user-images.githubusercontent.com/32570227/166705176-dd9d76b2-0944-4203-83ac-2a2712c8aad2.png)

I did get around the issue by creating a new managed property that I mapped the crawled property to my site column `<column_name>OWSURLH`. Then I have added that managed property to my selected properties and added a new Layout slot called Link.
![chrome_Qt61R8RXoW](https://user-images.githubusercontent.com/32570227/166705193-0cbf0f02-28c2-44fe-9e02-2178931d5ff8.png)
![chrome_plKvscoSUa](https://user-images.githubusercontent.com/32570227/166705200-eedf8839-9b7f-4713-b627-ca61bd69b88f.png)

Then, when displaying the items, I have the following code: `<a href="{{slot item @root.slots.Link}}" target="_blank" style="color:{{@root.theme.semanticColors.link}}" data-interception="off">{{getUrlField Link0OWSURLH "Title"}}</a>`

It now works exactly as I needed, thank you so much for your help! 
Kind regards, 
Yvett

**Reply by wobba** [reply]
`{{getUrlField Link0OWSURLH "URL"}}`  should work to get the URL as it splits the value on comma. Curious as to why you're getting the title when doing this.

**Reply by zkent** [reply]
If I don't see a managed property for my link column, do I just need to have someone with the correct permissions create a new managed property?

EDIT: To answer my own question, yes.  Create a managed property OR use a site column instead. For some reason, hyperlink fields require this when other field types do not.

**Reply by kasperbolarsen** [reply]
@zkent please create a new discussion item as this case is closed, thank you :-)

**Comment by tedtninety** [comment]
Hi guys, thank you so much for this.
This uses a "Details List" Layout as an example and I am using a "Custom" Layout and I had to make sense on how to apply accordingly.

As FranckyC and Wobba mentioned:

1 - Go to "Selected Properties"
add in   Column Name and OWSURLH .
So if your column name is "Link" then it would be LinkOWSURLH 

2- Now you can call the link's title name by adding {{getUrlField Link0OWSURLH "Title"}}

And now you have the URL Title for the URL's Alternate Text.
Hope this helps someone else who may need it.

**Reply by taliamath** [reply]
Hello, this thread has been helpful, but I need to display a hyperlink when a result is an URL. Most of the results are documents, but when I set my page up to display a field for the URL, it also shows up when the result is a document. It is given the name "undefined." How do I only show the slot when the result is an URL content type and not a document, and how do I display the URL as the text for the hyperlink instead of "undefined"?

**Reply by taliamath** [reply]
What I'd really like to do is change the title to link to an external website instead of downloading a file with the URL. Presently, I have results set up so that when you click the title, it opens the PDF in another window. The problem is that when the result is an URL, when you click the title, it downloads a file with the URL. I want the title to open the external website when clicked if the result is an URL, and to open the PDF in a new window when it is a document (usually a PDF).

**Reply by wobba** [reply]
@taliamath please log a separate question for your scenario and setup.
